huaweicloud/services/evs/data_source_huaweicloud_evs_volumes.go

```diff
@@ -0,0 +1,242 @@
+	return cloudvolumes.ListOpts{
+		AvailabilityZone:    d.Get("availability_zone").(string),
+		EnterpriseProjectID: d.Get("enterprise_project_id").(string),
+		Multiattach:         d.Get("shareable").(bool),
+		ServerID:            d.Get("server_id").(string),
+		Status:              d.Get("status").(string),
+	}
+}
+
+func sourceEvsAttachment(attachements []cloudvolumes.Attachment, mode string) []map[string]interface{} {
+	result := make([]map[string]interface{}, len(attachements))
+	for i, attachement := range attachements {
+		result[i] = map[string]interface{}{
+			"id":            attachement.AttachmentID,
+			"attached_at":   attachement.AttachedAt,
+			"attached_mode": mode,
+			"device_name":   attachement.Device,
+			"server_id":     attachement.ServerID,
+		}
+	}
+	return result
+}
+
+func sourceEvsVolumes(volumes []interface{}) ([]map[string]interface{}, []string, error) {
+	result := make([]map[string]interface{}, len(volumes))
+	ids := make([]string, len(volumes))
+
+	for i, val := range volumes {
+		volume := val.(cloudvolumes.Volume)
+		vMap := map[string]interface{}{
+			"id":                    volume.ID,
+			"attachments":           sourceEvsAttachment(volume.Attachments, volume.Metadata.AttachedMode),
+			"availability_zone":     volume.AvailabilityZone,
+			"description":           volume.Description,
+			"enterprise_project_id": volume.EnterpriseProjectID,
+			"shareable":             volume.Multiattach,
+			"name":                  volume.Name,
+			"service_type":          volume.ServiceType,
+			"size":                  volume.Size,
+			"status":                volume.Status,
+			"create_at":             volume.CreatedAt,
+			"update_at":             volume.UpdatedAt,
+			"tags":                  volume.Tags,
+			"wwn":                   volume.WWN,
+		}
+		bootable, err := strconv.ParseBool(volume.Bootable)
+		if err != nil {
+			return nil, nil, fmtp.Errorf("The bootable of volume (%s) connot be converted from boolen to string.",
+				volume.ID)
+		} else {
+			vMap["bootable"] = bootable
+		}
+
+		result[i] = vMap
+		ids[i] = volume.ID
+	}
+	return result, ids, nil
+}
+
+func dataSourceEvsVolumesV2Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*config.Config)
+	client, err := config.BlockStorageV2Client(config.GetRegion(d))
+	if err != nil {
+		return fmtp.DiagErrorf("Error creating HuaweiCloud EVS v2 client: %s", err)
+	}
+
+	pages, err := cloudvolumes.List(client, buildQueryOpts(d)).AllPages()
+	if err != nil {
+		return fmtp.DiagErrorf("An error occurred while fetching the pages of the EVS disks: %s", err)
+	}
+	volumes, err := cloudvolumes.ExtractVolumes(pages)
+	if err != nil {
+		return fmtp.DiagErrorf("Error getting the EVS volume list form server: %s", err)
+	}
+
+	filter := map[string]interface{}{
+		"Tags": d.Get("tags"),
+	}
+	filterVolumes, err := utils.FilterSliceWithField(volumes, filter)
```

Comment:
`tags` should be fuzzy matching, `utils.FilterSliceWithField` does not support
suggest to remove the filter of `tags`

Reply:
After the public function `HasMapContains` supports, `tags` supports fuzzy matching.